src/materialized/src/lib.rs

```diff
@@ -45,6 +45,7 @@ use crate::mux::Mux;
 
 mod http;
 mod mux;
+mod prof;
```

Comment:
I know it's not crazy right now, but I would still prefer if this logic was all in its own crate, instead of in a module. It makes the API contract much clearer and generally would allow us to configure it in other crates more easily, and breaking this stuff out into its own crate just gets harder to do as it gets more entwined in the current crate.

Relatedly, below I say that this whole module should be cfg-dependent on the target_os being macos, and if you have an optional crate then you don't build any of its dependencies unless you are actually building the crate. It's harder to do things correctly like that if you're working on just a module.